docs/DataTypes.md

```diff
@@ -0,0 +1,40 @@
+# Data Types
+
+DataHelix currently recognises three distinct data types. Keeping this set small is a deliberate goal; it would be possible to have types like _FirstName_ or _Postcode_, but instead these are considered specialisations of the _String_ type, so they can be constrained by the normal string operators (e.g. a user could generate all first names shorter than 10 characters, starting with a vowel).
+
+## Numeric
+
+In principle, any real number. In practice, any number that can be represented in a Java [BigDecimal](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html) object.
+
+In profile files, numbers must be expressed as JSON numbers, without quotation marks.
+
+### Granularity
+
+The granularity of a numeric field is a measure of how small the distinctions in that field can be; it is the smallest positive number of which every valid value is a multiple. For instance:
+
+- if a numeric field has a granularity of 1, it can only be satisfied with multiples of 1, a.k.a. integers
+- if a numeric field has a granularity of 0.1, it can be satified by 1, or 1.1, but not 1.11
+
+At present, only granularities less than or equal to 1 are supported (in future, it may be possible to specify higher granularities - a granularity of 2 would admit only even numbers). Additionally, granularities must be powers of ten, so 0.01 is legal but 0.02 is not.
```

Comment:
This description makes it sound like granularity sets generation to a level of decimal points and works up to an integer (i.e. a granularity of 0.001 would create numbers at 3dp / 2dp / 1dp and 0dp) is that correct?

How would I lock in a granularity of 2dp only, for currencies?

Does granularity only effect values that end with a 1?

Reply:
You're mixing two different concepts. 0.1 and 0.10 are the same number and have the same granularity. If you're outputting to a text format and want the latter rather than the former, you need a presentational constraint - ie, `formattedAs`. IIRC there's something in the FAQ about the difference between the two - it would be good to link there from this documentation.

Reply:
If the concepts from this description could be mixed incorrectly, is there a way that we can make this clearer?

Reply:
Agreed, let's link it so that people won't incorrectly mix these concepts.

Reply:
Done 👍